letsencrypt/client/init.sls

```diff
@@ -107,4 +107,21 @@ certbot_cron:
 
 {%- endif %}
 
+{%- for hookset, hooks in client.get("hooks", {}).items() %}
+{%- for hook in hooks %}
+{#- FIXME: Should probably complain if something other than
+    pre/post/deploy is given, but I'm not sure how. #}
+
+certbot_renewal_{{ hookset }}_hook_{{ loop.index }}:
+  file.managed:
+    - name: /etc/letsencrypt/renewal-hooks/{{ hookset }}/{{ hook.split("/") | last }}
```

Comment:
I think you should be able to supply whole scripts in model and not only source on salt filesystem.

Reply:
I'm not sure I understand the request. Could you give an example?

Reply:
I would suggest using metadata like this:

```
letsencrypt:
  client:
    hook:
      pre:
        # Use salt fileserver
        - source: salt://path/to/prehook1.sh
        # Or generate file by script as a content
        - script: |
            #!/bin/bash
            echo "Triggered pre-hook"

So you can supply scripts on salt filesystem or pass them directly.

Reply:
Putting scripts directly in pillar fills me with fear and loathing, but I don't really have a good argument against supporting it except perhaps that it would make the state more complicated. I'll see what I can do and then update the PR.